tls/s2n_crl.c

```diff
@@ -91,6 +91,47 @@ int s2n_crl_get_issuer_hash(struct s2n_crl *crl, uint64_t *hash)
     return S2N_SUCCESS;
 }
 
+int s2n_crl_validate_this_update(struct s2n_crl *crl)
+{
+    POSIX_ENSURE_REF(crl);
+    POSIX_ENSURE_REF(crl->crl);
+
+    ASN1_TIME *this_update = X509_CRL_get_lastUpdate(crl->crl);
+    POSIX_ENSURE_REF(this_update);
+
+    int ret = X509_cmp_time(this_update, NULL);
+    if (ret == 0) {
+        POSIX_BAIL(S2N_ERR_CRL_INVALID_THIS_UPDATE);
+    }
+    if (ret > 0) {
+        POSIX_BAIL(S2N_ERR_CRL_NOT_YET_VALID);
+    }
+
+    return S2N_SUCCESS;
+}
+
+int s2n_crl_validate_next_update(struct s2n_crl *crl)
+{
+    POSIX_ENSURE_REF(crl);
+    POSIX_ENSURE_REF(crl->crl);
+
+    ASN1_TIME *next_update = X509_CRL_get_nextUpdate(crl->crl);
+    if (next_update == NULL) {
+        /* If the CRL has no nextUpdate field, assume it will never expire */
+        return S2N_SUCCESS;
```

Comment:
Does Openssl also make this assumption? Like, is this the standard meaning?

Reply:
Yes, this is what OpenSSL does. They don't fail `check_crl_time` if nextUpdate doesn't exist.

https://github.com/openssl/openssl/blob/00e38edcfb95b556a59de96e0c18343828929c8f/crypto/x509/x509_vfy.c#L1045-L1059

Reply:
Not blocking, but you might want to do some quick research to see if that's officially defined anywhere.
The first RFCs I found seem ot think that newUpdate is mandatory:
- https://datatracker.ietf.org/doc/html/rfc2459#section-5.1.2.5
- https://datatracker.ietf.org/doc/html/rfc5280#section-5.1.2.5
But maybe those aren't the right RFCs?

Reply:
I think that RFC (5280) is correct. It does say it's mandatory, however, it also says this:

`The behavior of clients processing CRLs that omit
   nextUpdate is not specified by this profile.`

So I suppose OpenSSL just decides to ignore it if it's not there. Is this reasonable?

Reply:
Bouncy castle also does this: https://github.com/bcgit/bc-java/blob/bff7fd2c3a55c4ed82b4a25de74f2583577116a4/pkix/src/main/java/org/bouncycastle/pkix/jcajce/PKIXCRLUtil.java#L42-L51